TrackingTools/RecoGeometry/src/GlobalDetLayerGeometry.cc

```diff
@@ -12,7 +12,7 @@ const DetLayer* GlobalDetLayerGeometry::idToLayer(const DetId& detId) const {
   else {
     throw cms::Exception("DetLayers")
         << "Error: called GlobalDetLayerGeometry::idToLayer() for a detId which is neither Tracker nor Muon "
-        << (mtd_ == nullptr ? "" : "nor MTD ") << detId;
+        << (mtd_ == nullptr ? "" : "nor MTD ") << " det rawId " << detId.rawId() << " subdetId " << detId.subdetId();
```

Comment:
@parbol you might possibly add also ```detId.det()``` for a more straightforward information

Reply:
Hi Fabio,

Yes, I have included this in a later commit.